collab-document/tests/document/redo_undo_test.rs

```diff
@@ -0,0 +1,201 @@
+use nanoid::nanoid;
+use serde_json::to_value;
+use std::collections::HashMap;
+use std::time::Duration;
+
+use crate::util::{
+  create_document, create_document_with_db, db, delete_block, insert_block_for_page,
+  open_document_with_db, update_block,
+};
+
+const WAIT_TIME: Duration = Duration::from_secs(1);
+
+#[tokio::test]
+async fn insert_undo_redo() {
+  let doc_id = "1";
+  let test = create_document(1, doc_id);
+  tokio::time::sleep(Duration::from_secs(1)).await;
+  let document = test.document;
+  let block_id = nanoid!(10);
+
+  insert_block_for_page(&document, block_id.clone());
```

Comment:
Option: Return the inserted block from this function, and then we can use it to check if the inserted block is equal to the redo block. This will make the test more reliable.

Reply:
This test is in `block_test`. Don't confuse test coverage for different tests